commands/new_function.go

```diff
@@ -346,3 +351,33 @@ Cannot have duplicate function names in same yaml file`, functionName, appendFil
 
 	return nil
 }
+
+func addLastNewline(fileName string) error {
+	// open a file as read write
+	file, err := os.OpenFile(fileName, os.O_RDWR, 0600)
+	defer file.Close()
+	if err != nil {
+		return fmt.Errorf("could not open '%s' to check for new lines %s", fileName, err)
+	}
+
+	// read the last byte of the file (excludes EOF)
+	buffer := make([]byte, 1)
+	fileStats, _ := file.Stat()
+	bytesRead, err := file.ReadAt(buffer, fileStats.Size()-1)
+
+	// handle I/O errors
+	if err != nil {
+		return fmt.Errorf("could not read the last byte of '%s' %s", fileName, err)
+	} else if bytesRead != 1 {
+		return fmt.Errorf("read unexpected # of bytes in '%s'", fileName)
+	}
+
+	hasTrailingNewline := string(buffer) == "\n"
+	if !hasTrailingNewline {
+		// add 2 trailing newlines
+		// this is consistent with append behavior when last byte is already \n
+		file.Seek(0, 2)
+		file.Write([]byte("\n\n"))
```

Comment:
Some OSes may use `\r\n`https://en.wikipedia.org/wiki/Newline

I.e LF vs CR LF

Reply:
:+1:  I thought the same thing, and bumped [into this when searching](https://stackoverflow.com/a/14494187/326313), and if you [look at line 1171](https://golang.org/src/fmt/print.go) here, you'll see even the std lib doesn't write `\r\n` vs `\n`. :exploding_head: 

I'm not sure how to proceed. Where else do we write `\r\n` versus `\n`, or is there a library/standard method that does this for us?

Reply:
PS: Thanks again for the review! :100:

Reply:
I'll setup a Windows VM and test with Git Bash to be safe. :) [Given this thread](https://github.com/golang/go/issues/28822#issuecomment-441430941), I assume Powershell and Git Bash can work with `\n`. Talk to you later.

Reply:
Works like a champ in Windows. In the below pic, **new2** and **new4** were appended when there were no trailing new lines. **new3** was appended when there were trailing new lines. 

![image](https://user-images.githubusercontent.com/1272076/142708962-0ec12d76-cc52-42f3-8a24-0e26b5fb05f6.png)

I'm using Powershell here to test, and assume I'd get the same result wiht git bash.

Reply:
I'll test with git bash tomorrow, and push unit tests.